### PR TITLE
Covert OnSwipeTouchListener.java to Kotlin

### LINKED ISSUE #2400: Convert OnSwipeTouchListener to kotlin
Is your feature request related to a problem? Please describe.

OnSwipeTouchListener is in Java

Describe the solution you'd like

Convert it in Kotlin

Additional context

Parent issue #1826

### PR DESCRIPTION
**Pull request by abdulwd**
Fixes #2400

**Screenshots**

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/OnSwipeTouchListener.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      return super.onSingleTapUp(event)
+    }
+
+    override fun onFling(
+      e1: MotionEvent,
+      e2: MotionEvent,
+      velocityX: Float,
+      velocityY: Float
+    ): Boolean {
+      var result = false
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/OnSwipeTouchListener.kt:55]: can we get rid of this variable? just return at the points of assignment

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/OnSwipeTouchListener.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+  }
+
+  open fun onSwipeRight() {}
+  open fun onSwipeLeft() {}
+  fun onSwipeTop() {}
+  open fun onSwipeBottom() {}
+  open fun onTap(e: MotionEvent?) {}
+
+  init {
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/OnSwipeTouchListener.kt:89]: this should probably be at the top

Comment (abdulwd) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/OnSwipeTouchListener.kt]: fdd9566ac897602fc37db9e8ecd5d0a2e6a729b9

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/OnSwipeTouchListener.kt (hunk 1 above)
Comment (abdulwd) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/OnSwipeTouchListener.kt]: fdd9566ac897602fc37db9e8ecd5d0a2e6a729b9

**Review by macgills — APPROVED** [review]
CodeFactor is complaining but I don't care too much as removing the method would make the class a bit less symmetrical

**Comment by abdulwd** [comment]
Yeah, I thought maybe we should leave it for any future changes we might want to make.

**Pull request merged**